### DISCUSSION TITLE
【求助】使用DS3.1.1+Flink1.17.0版本创建Flink Stream类型的工作流后无法产生任务实例

### DISCUSSION THREAD

**Discussion by licanxue** [apache/dolphinscheduler#14432] [Q&A]
使用DS3.1.1+Flink1.17.0版本创建Flink Stream类型的工作流后无法产生任务实例，我的工作流定义如下：
![image](https://github.com/apache/dolphinscheduler/assets/86290007/61fe328d-74be-456f-bbea-bedcaec85323)
![image](https://github.com/apache/dolphinscheduler/assets/86290007/73559202-1ccf-4e58-ae5f-cd30031e2ac9)
![image](https://github.com/apache/dolphinscheduler/assets/86290007/7b34d29a-82b0-4e05-9ec1-4d7312960a0b)
保存上线之后，点击运行，可以看到工作流实例中产生了对应的记录，但是运行时间只有1s
![image](https://github.com/apache/dolphinscheduler/assets/86290007/d2d232bc-846f-4397-84ae-a8156cd97e0d)
然后此时去任务实例的实时任务列表中，没有产生对应的任务实例：
![image](https://github.com/apache/dolphinscheduler/assets/86290007/211a8b04-a802-4f70-83e5-a84c442d5beb)
请问，这是什么原因？是否DS的版本和Flink不匹配？我创建Flink的任务却可以正常运行。

**Comment by misszhangss** [comment]
@licanxue 你好这个问题解决了吗  如果没有解决 可以尝试在任务定义中尝试创建flink_stream任务，但是会遇到你一个问题就是实时任务运行一会就自动结束了，如果您这边问题解决了或者执行实时任务不会自动结束的话请和我分享一下解决方案  谢谢
![image](https://github.com/apache/dolphinscheduler/assets/84503545/01c7f2fa-8c41-40be-a331-16d40b219858)

**Comment by 18772247265** [comment]
我也遇到这个问题，看了下工作流创建时候前端参数，里边有个taskExecuteType:batch，相当于说默认把任务执行类型设置为了批模式，这应该是个bug吧，也没有让自己选择type的地方，就默认写死了batch

**Comment by keepnk** [comment]
@licanxue 您好，1秒结束，没有任务实例的情况我也遇到了，您解决了？

**Comment by misszhangss** [comment]
我后续是通过修改源码解决的，启动flink是通过脚本启动的，修改那个脚本就可以解决这个问题




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "apache/dolphinscheduler"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月26日(星期四) 上午10:56
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [apache/dolphinscheduler] 【求助】使用DS3.1.1+Flink1.17.0版本创建Flink Stream类型的工作流后无法产生任务实例 (Discussion #14432)





 
@licanxue 您好，1秒结束，没有任务实例的情况我也遇到了，您解决了？

**Reply by keepnk** [reply]
您改的是sql-client.sh脚本么？
